dist/datasource.js

```diff
@@ -274,53 +281,144 @@ function (angular, _, dateMath, moment) {
+                var resultByQueryMetric = resultByQuery.data[j];
+                var metricName = resultByQueryMetric.target;
+
+                // default no group by
+                var resultsTitle = outputMetricName;
+
+                if (resultByQueryMetric.props) {
+                    var groupByValues = Object.values(resultByQueryMetric.props);
+
+                    if (groupByValues instanceof Array && groupByValues.length > 0) {
+                        resultsTitle = groupByValues[0];
+                    }
+                }
+
+                if (!!!resultsHashMap[resultsTitle]) {
+                    resultsHashMap[resultsTitle] = {};
+                }
+
+                var dstTitleData = resultsTotalList.find(x => x.id === resultsTitle);
+
+                if (!dstTitleData) {
+                    dstTitleData = {
+                        id: resultsTitle,
+                        value: []
+                    };
+                    resultsTotalList.push(dstTitleData);
+                }
+
+                var resultsHash = resultsHashMap[resultsTitle];
+
+                if (resultByQueryMetric.datapoints) {
+                    var metricMap = {};
+                    metricMap[metricName] = 0;
+
+                    for (var k = 0; k < resultByQueryMetric.datapoints.length; k++) {
+                        var datapoint = resultByQueryMetric.datapoints[k];
+                        resultsHash[datapoint[1]] = resultsHash[datapoint[1]] || [];
+                        resultsHash[datapoint[1]][i] = resultsHash[datapoint[1]][i] || {};
+                        resultsHash[datapoint[1]][i][metricName] = datapoint[0];
+                        metricMap[metricName] += datapoint[0];
+                    }
+
+                    dstTitleData.value.push(metricMap);
+                }
+            }
+        }
+
+        var orderType = target.orderType;
+        var orderSize = target.orderSize;
+        var resultTitles = Object.keys(resultsHashMap);
+
+        if (!(resultTitles.length === 1 && resultTitles[0] === outputMetricName)) {
+            resultTitles = [];
+            // compute by group name
+            resultsTotalList.forEach(function(resultTotal) {
+                var result = 0;
+
+                try {
+                    result = expressionFunction.apply(this, resultTotal.value);
+                } catch(err) {
+                    console.log(err);
+                }
+
+                resultTotal.value = result;
+            });
+
+            // sort results
+            resultsTotalList.sort(function(obj1, obj2) {
+                var val1 = obj1.value;
+                var val2 = obj2.value;
+
+                if (val1 < val2) {
+                    return -1;
+                } else if (val1 > val2) {
+                    return 1;
+                } else {
+                    return 0;
+                }
+            });
+
+            if (orderSize > resultsTotalList.length) {
```

Comment:
This code is not working as intended when you pick "No limit" as your group by size. 

```suggestion
            if (orderSize > resultsTotalList.length || orderSize == 0) {
```

This should fix it.

Reply:
Other than this it seems to be working as intended to me (I'm using it only on elasticsearch datasource)

Reply:
Yeah，I have fixed the bug in my project.